spec.emu

```diff
@@ -0,0 +1,1437 @@
+    <dd>
+      <p>zero-based indexed offset within a line of the generated code, computed as UTF-16 code units for JavaScript and CSS source maps, and as byte indexes in the binary content (represented as a single line) for WebAssembly source maps.</p>
+      <emu-note>That means that "A" (`LATIN CAPITAL LETTER A`) measures as 1 code unit, and "🔥" (`FIRE`) measures as 2 code units. Source maps for other content types may diverge from this.</emu-note>
+    </dd>
+  </dl>
+</emu-clause>
+
+<emu-clause id="sec-json-values-utilities">
+  <h1>JSON values utilities</h1>
+
+  <p>While this specification's algorithms are defined on top of ECMA-262 internals, it is meant to be easily implementable by non-JavaScript platforms. This section contains utilities for working with JSON values, abstracting away ECMA-262  details from the rest of the document.</p>
+
+  <p>A <dfn id="type-json-value" variants="JSON values">JSON value</dfn> is either a JSON object, a JSON array, a <emu-xref href="#sec-ecmascript-language-types-string-type">String</emu-xref>, a <emu-xref href="#sec-ecmascript-language-types-number-type">Number</emu-xref>, a <emu-xref href="#sec-ecmascript-language-types-boolean-type">Boolean</emu-xref>, or <emu-xref href="#sec-ecmascript-language-types-null-type">*null*</emu-xref>.</p>
+
+  <p>A <dfn id="type-json-object">JSON object</dfn> is an Object such that each of its properties:</p>
+  <ul>
+    <li>is a data property,</li>
+    <li>has a String key,</li>
+    <li>has JSON value value.</li>
+  </ul>
+
+  <p>A <dfn id="type-json-array">JSON array</dfn> is a JSON object such that:</p>
+  <ul>
+    <li>it has a property whose key is *"length"* and whose value is a Number,</li>
+    <li>all its other properties' keys are integer indices.</li>
+  </ul>
+
+  <emu-clause id="sec-ParseJSON" type="abstract operation">
+    <h1>
+      ParseJSON (
+        _string_: a String,
+      ): a JSON value
+    </h1>
+    <dl class="header"></dl>
+    <emu-alg>
+      1. Let _result_ be Call(<emu-xref href="#sec-json.parse">%JSON.parse%</emu-xref>, *null*, « _string_ »).
+      1. Assert: _result_ is a JSON value.
+      1. Return _result_.
+    </emu-alg>
+
+    <emu-note type="editor">
+      This abstract operation is in the process of being exposed by ECMA-262 itself, at <a href="https://github.com/tc39/ecma262/pull/3540">tc39/ecma262#3540</a>.
+    </emu-note>
+  </emu-clause>
+
+  <emu-clause id="sec-JSONObjectGet" type="abstract operation">
+    <h1>
+      JSONObjectGet (
+        _object_: a JSON object,
+        _key_: a String,
+      ): a JSON value or ~missing~
+    </h1>
+    <dl class="header">
+      <dt>description</dt>
+      <dd>It returns the value associated with the specified _key_ in _object_.</dd>
+    </dl>
+    <emu-alg>
+      1. If _object_ does not have an own property with key _key_, return ~missing~.
+      1. Let _prop_ be _object_'s own property whose key is _key_.
+      1. Return _prop_.[[Value]].
```

Comment:
This might be overly nitpicky, but I think `[[Value]]` here is technically an attribute rather than field so should we do something like `Return _prop_'s [[Value]] attribute`?